environment-api/src/main/java/com/sequenceiq/environment/api/v1/platformresource/CredentialPlatformResourceEndpoint.java

```diff
@@ -191,4 +192,17 @@ PlatformResourceGroupsResponse getResourceGroups(
             @QueryParam("region") String region,
             @QueryParam("platformVariant") String platformVariant,
             @QueryParam("availabilityZone") String availabilityZone);
+
+    @GET
+    @Path("private_dns_zones")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = OpDescription.GET_PRIVATE_DNS_ZONES, produces = MediaType.APPLICATION_JSON, notes = CONNECTOR_NOTES,
+            nickname = "getPrivateDnsZones")
+    PlatformPrivateDnsZonesResponse getPrivateDnsZones(
+            @QueryParam("credentialName") String credentialName,
+            @QueryParam("credentialCrn") String credentialCrn,
+            @QueryParam("region") String region,
+            @QueryParam("platformVariant") String platformVariant,
+            @QueryParam("availabilityZone") String availabilityZone);
```

Comment:
is there an explicit need for `availabilityZone` here? If yes, what's that?

Reply:
I agree with all your comments regarding the unnecessary query parameters. I think they would also apply for the getResourceGroups request as well (we do not use neither region nor availability zone info). I suspected that the unneeded parameters are there for uniformity, but I am glad to remove them, as adding them later does not break the api.

Reply:
I asked Wolf team about the unnecessary parameters, and the response was that the endpoints need to serve all platforms. True that privateDnsZones are needed only on Azure now. But private DNS zones actually exist also for GCP (https://cloud.google.com/blog/products/networking/introducing-private-dns-zones-resolve-to-keep-internal-networks-concealed) and aws (private hosted zones: https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/hosted-zones-private.html). Given this, I would not change the endpoints, nor would add javadoc that some query parameters are not used.

Reply:
I agree, but to my knowledge, none of those are `availabilityZone`-aware, but VPC, so it should not be there, if it is not needed

GCP: https://cloud.google.com/dns/docs/zones#create-pub-zone
AWS: https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/hosted-zone-private-creating.html

Region should be checked, too

Reply:
Ok, as agreed with @biharitomi, since DNS zones are global on all 3 platforms (AWS, Azure, GCP), both region and availability zones were deleted from the endpoint. Also, if it would be needed for some reason, it is not a breaking change to put it back.